openquake/calculators/scenario_risk.py

```diff
@@ -73,11 +74,10 @@ def scenario_risk(riskinputs, riskmodel, rlzs_assoc, monitor):
             stats[:, r, 1] = out.loss_matrix.std(ddof=1, axis=1)
             stats[:, r, 2] = out.insured_loss_matrix.mean(axis=1)
             stats[:, r, 3] = out.insured_loss_matrix.std(ddof=1, axis=1)
-            avg = result[lti]
             for asset, stat in zip(out.assets, stats):
-                avg['avg', r, asset.idx] = stat
-            result[lti]['agg', r, 0] = out.aggregate_losses
-            result[lti]['agg', r, 1] = out.insured_losses
+                result['avg'].append((l, r, asset.idx, stat))
+            result['agg'][:, l, r, 0] += out.aggregate_losses
+            result['agg'][:, l, r, 1] += out.insured_losses
     return result
```

Comment:
Please check if the nature of the return value mentioned in the docstring matches the actual value returned by this function.